utils/utils.go

```diff
@@ -118,14 +119,13 @@ func IsValidEthereumAddress(address string) bool {
 	return ethAddrPattern.MatchString(address)
 }
 
-func ReadPublicURL(url string) ([]byte, error) {
-	// allow no redirects
-	httpClient := http.Client{
-		CheckRedirect: func(req *http.Request, via []*http.Request) error {
-			return http.ErrUseLastResponse
-		},
-		Timeout: 3 * time.Second,
+func ReadPublicURL(url string, httpClient *http.Client) ([]byte, error) {
```

Comment:
Had to change the signature to make it possible to mock. IMO we shouldn't do real HTTP reqs from tests.